lang/cs/Org.Apache.REEF.IMRU/OnREEF/CheckpointHandler/IMRUCheckpointHandler.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.IO;
+using System.Linq;
+using Org.Apache.REEF.IMRU.API;
+using Org.Apache.REEF.IMRU.OnREEF.IMRUTasks;
+using Org.Apache.REEF.IO.FileSystem;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake.Remote;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.CheckpointHandler
+{
+    /// <summary>
+    /// Default implementation of IIMRUCheckpointHandler
+    /// </summary>
+    public class IMRUCheckpointHandler : IIMRUCheckpointHandler
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(IMRUCheckpointHandler));
+
+        private readonly IFileSystem _fileSystem;
+        private readonly string _checkpointFilePath;
+        private const string StateDir = "StateDir";
+        private const string StateFile = "StateFile";
+        private const string FlagFile = "FlagFile";
+        public const string StateFileExt = ".bin";
+        public const string FlagFileExt = ".txt";
+
+        /// <summary>
+        /// It is for storing and retrieving checkpoint data.
+        /// </summary>
+        /// <param name="checkpointFilePath">The file path where the checkpoint data will be stored.</param>
+        /// <param name="fileSystem">File system to load/upload checkpoint data</param>
+        [Inject]
+        private IMRUCheckpointHandler(
+            [Parameter(typeof(CheckpointFilePath))] string checkpointFilePath,
+            IFileSystem fileSystem)
+        {
+            _fileSystem = fileSystem;
+            _checkpointFilePath = checkpointFilePath;
+
+            Logger.Log(Level.Info, "State file path: {0}", checkpointFilePath);
+        }
+
+        /// <summary>
+        /// Save serialized checkpoint data to remote checkpoint file.
+        /// </summary>
+        /// <param name="taskState"></param>
+        /// <param name="codec"></param>
+        public void Persistent(ITaskState taskState, ICodec<ITaskState> codec)
```

Comment:
the verb would be `Persist`. Alternatively, call it `save` or `store`.

Reply:
done